vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gatt_server.c

```diff
@@ -856,7 +858,17 @@ BTStatus_t prvAddServiceBlob( uint8_t ucServerIf,
         if( serviceCnt < MAX_SERVICES )
         {
             pSvc = &espServices[ serviceCnt ];
-            afrServices[ serviceCnt ] = pxService;
+            afrFullService = pvPortCalloc( 1, sizeof( BTService_t ) );
```

Comment:
@prasad-alatkar Could you explain why an internal copy was required. As far as my understanding these pointers are to static memory allocated throughout program context.

Reply:
@ravibhagavandas , the original request came from DPK team regarding this. Though the pointers are pointing to static memory (i.e. `afrServices`), but the array `afrServices` points to `pxService` received from upper layer in `prvAddServiceBlob`. Now, the upper layer may free this memory at their will (apparently the case with DPK), so to avoid dependency on upper layer I believe this PR is needed. Please let me know your thoughts on this.